### PR TITLE
Add table oci_core_load_balancer. Closes #182

### LINKED ISSUE #182: Add table oci_core_load_balancer
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by rajeshbal65**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Test names: [ 'tests/oci_core_load_balancer' ]
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_core_load_balancer []

PRETEST: tests/oci_core_load_balancer

TEST: tests/oci_core_load_balancer
Running terraform
oci_core_vcn.named_test_resource: Refreshing state... [id=ocid1.vcn.oc1.ap-mumbai-1.amaaaaaa6igdexaa5haionftcongwv73sggdnl4u74z5hfowtbhel2jvi6aa]
oci_core_subnet.named_test_resource: Refreshing state... [id=ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaaspyds2srtmoou3phgtzrejy36ozxls3hl6fqtcxnohqdxl3sto6q]
oci_load_balancer_load_balancer.named_test_resource: Refreshing state... [id=ocid1.loadbalancer.oc1.ap-mumbai-1.aaaaaaaaqgldtfprud77wdlsxddpukrj7qmm6assvsh2iaa752ps37hqirga]
oci_core_vcn.named_test_resource: Modifying... [id=ocid1.vcn.oc1.ap-mumbai-1.amaaaaaa6igdexaa5haionftcongwv73sggdnl4u74z5hfowtbhel2jvi6aa]
oci_core_vcn.named_test_resource: Modifications complete after 1s [id=ocid1.vcn.oc1.ap-mumbai-1.amaaaaaa6igdexaa5haionftcongwv73sggdnl4u74z5hfowtbhel2jvi6aa]
oci_core_subnet.named_test_resource: Modifying... [id=ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaaspyds2srtmoou3phgtzrejy36ozxls3hl6fqtcxnohqdxl3sto6q]
oci_core_subnet.named_test_resource: Modifications complete after 0s [id=ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaaspyds2srtmoou3phgtzrejy36ozxls3hl6fqtcxnohqdxl3sto6q]
oci_load_balancer_load_balancer.named_test_resource: Creating...
oci_load_balancer_load_balancer.named_test_resource: Still creating... [10s elapsed]
oci_load_balancer_load_balancer.named_test_resource: Still creating... [20s elapsed]
oci_load_balancer_load_balancer.named_test_resource: Still creating... [30s elapsed]
oci_load_balancer_load_balancer.named_test_resource: Creation complete after 35s [id=ocid1.loadbalancer.oc1.ap-mumbai-1.aaaaaaaatet2ki6gaoxrqm2opjsum3ipc2piw5t7oobfehooarjc7pyjfrta]

Apply complete! Resources: 1 added, 2 changed, 0 destroyed.

Outputs:

display_name = "steampipetest3152"
ip_address_details = tolist([
  {
    "ip_address" = "152.67.23.194"
    "is_public" = true
    "reserved_ip" = tolist([])
  },
])
is_private = false
region = "ap-mumbai-1"
resource_id = "ocid1.loadbalancer.oc1.ap-mumbai-1.aaaaaaaatet2ki6gaoxrqm2opjsum3ipc2piw5t7oobfehooarjc7pyjfrta"
resource_name = "steampipetest3152"
shape = "100Mbps"
state = "ACTIVE"
subnet_ids = tolist([
  "ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaaspyds2srtmoou3phgtzrejy36ozxls3hl6fqtcxnohqdxl3sto6q",
])
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"

Running SQL query: test-get-query.sql

[
  {
    "display_name": "steampipetest3152",
    "id": "ocid1.loadbalancer.oc1.ap-mumbai-1.aaaaaaaatet2ki6gaoxrqm2opjsum3ipc2piw5t7oobfehooarjc7pyjfrta",
    "is_private": false,
    "lifecycle_state": "ACTIVE",
    "shape_name": "100Mbps"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "display_name": "steampipetest3152",
    "id": "ocid1.loadbalancer.oc1.ap-mumbai-1.aaaaaaaatet2ki6gaoxrqm2opjsum3ipc2piw5t7oobfehooarjc7pyjfrta",
    "is_private": false,
    "lifecycle_state": "ACTIVE",
    "shape_name": "100Mbps"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "region": "ap-mumbai-1",
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "steampipetest3152"
  }
]
✔ PASSED

POSTTEST: tests/oci_core_load_balancer

TEARDOWN: tests/oci_core_load_balancer


Error: 409-Conflict
Provider version: 4.36.0, released on 2021-07-21.
Service: Core Subnet
Error Message: The Subnet ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaaspyds2srtmoou3phgtzrejy36ozxls3hl6fqtcxnohqdxl3sto6q references the VNIC ocid1.vnic.oc1.ap-mumbai-1.abrg6ljrhddxwszv7jvc37q26wz6n7ecxlkkwbetxy2h57gc5tqifkd3okba. You must remove the reference to proceed with this operation.
OPC request ID: ab9f9e6d9ef6e49e97d2a3183f7e69cc/530D4A0E25E629BBC26CE562114D6AC8/4FDB7E61C3C7A5BE2BE32674F56E85CA
Suggestion: The resource is in a conflicted state. Please retry again or contact support for help with service: Core Subnet



SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  display_name,
  id,
  time_created,
  lifecycle_state as state,
  shape_name
from
  oci_core_load_balancer;

+-------------------+-------------------------------------------------------------------------------------------------+---------------------+--------+------------+
| display_name      | id                                                                                              | time_created        | state  | shape_name |
+-------------------+-------------------------------------------------------------------------------------------------+---------------------+--------+------------+
| lb_2021-0722-1300 | ocid1.loadbalancer.oc1.ap-mumbai-1.aaaaaaaakmxbd4cyjra6ijdg4hygg2b3ztigx6zx2mdq2cbxsvm7bdomjn2q | 2021-07-22 07:31:41 | ACTIVE | flexible   |
| steampipetest9204 | ocid1.loadbalancer.oc1.ap-mumbai-1.aaaaaaaatktds4opwj6fud6ey4kcbw2zywvmqxfptv6hutzqdfhy5c7kb2ba | 2021-07-22 11:11:52 | ACTIVE | 100Mbps    |
+-------------------+-------------------------------------------------------------------------------------------------+---------------------+--------+------------+
```

### List load balancers assigns with public IP address

```sql
select
  display_name,
  id,
  is_private
from
  oci_core_load_balancer
where
  not is_private;

+-------------------+-------------------------------------------------------------------------------------------------+------------+
| display_name      | id                                                                                              | is_private |
+-------------------+-------------------------------------------------------------------------------------------------+------------+
| lb_2021-0722-1300 | ocid1.loadbalancer.oc1.ap-mumbai-1.aaaaaaaakmxbd4cyjra6ijdg4hygg2b3ztigx6zx2mdq2cbxsvm7bdomjn2q | false      |
| steampipetest9204 | ocid1.loadbalancer.oc1.ap-mumbai-1.aaaaaaaatktds4opwj6fud6ey4kcbw2zywvmqxfptv6hutzqdfhy5c7kb2ba | false      |
+-------------------+-------------------------------------------------------------------------------------------------+------------+
```
</details>

### CODE REVIEW

File: oci/table_oci_core_load_balancer.go [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+	logger.Debug("oci.getCoreLoadBalancer", "Compartment", compartment, "OCI_REGION", region)
+
+	// Rstrict the api call to only root compartment/ per region
+	if !strings.HasPrefix(compartment, "ocid1.tenancy.oc1") {
+		return nil, nil
+	}
+
+	id := d.KeyColumnQuals["id"].GetStringValue()
+
+	if strings.TrimSpace(id) == "" {
```
Comment (bigdatasourav) [inline oci/table_oci_core_load_balancer.go:238]:
```suggestion
        // handle empty id in get call
	if strings.TrimSpace(id) == "" {
```

File: oci/table_oci_core_load_balancer.go [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+//// HYDRATE FUNCTION
+
+func getCoreLoadBalancer(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getCoreLoadBalancer")
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Debug("oci.getCoreLoadBalancer", "Compartment", compartment, "OCI_REGION", region)
+
+	// Rstrict the api call to only root compartment/ per region
```
Comment (bigdatasourav) [inline oci/table_oci_core_load_balancer.go:231]:
```suggestion
	// Restrict the api call to only root compartment/ per region
```

File: oci/table_oci_core_load_balancer.go [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+		return nil, err
+	}
+
+	return response.LoadBalancer, nil
+}
+
+//// TRANSFORM FUNCTION
+
+func loadBalancerTags(_ context.Context, d *transform.TransformData) (interface{}, error) {
+	loadBalancer := d.HydrateItem.(loadbalancer.LoadBalancer)
```
Comment (bigdatasourav) [inline oci/table_oci_core_load_balancer.go:266]: Need to add System tags as well.

**Review by bigdatasourav — COMMENTED** [review]
@rajeshbal65 plz check the comments also sort the column order after top columns.

**Review by bigdatasourav — CHANGES_REQUESTED** [review]
@rajeshbal65 please check the suggested changes.

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**